modules/aggregation-layers/src/utils/cpu-aggregator.js

```diff
@@ -140,18 +140,22 @@ export default class CPUAggregator {
     return defaultDimensions;
   }
 
-  updateState({oldProps, props, changeFlags}, viewport) {
+  updateState(opts) {
+    const {oldProps, props, changeFlags, viewport, attributes, projectPoints} = opts;
     this.updateGetValueFuncs(oldProps, props, changeFlags);
     const reprojectNeeded = this.needsReProjectPoints(oldProps, props, changeFlags);
-
+    let aggregationDirty = false;
     if (changeFlags.dataChanged || reprojectNeeded) {
-      // project data into hexagons, and get sortedColorBins
-      this.getAggregatedData(props, viewport);
+      // project data into bin and aggregate wegiths per bin
+      this.getAggregatedData(props, viewport, attributes, projectPoints);
```

Comment:
Naming: `get` needs to return something, while `update` changes internal states.